commit 6665d4a25da6dc1788010466f6f52e1df94a048b
Author: Steve McIntyre <steve.mcintyre@linaro.org>
Date:   Mon Nov 19 01:12:53 2012 -0500

    Add ldconfig cache tag handling for ARM hard-float ABI
    
    	* sysdeps/generic/ldconfig.h (FLAG_ARM_LIBHF): New macro.
    	* elf/cache.c (print_entry): Print ",hard-float" for
    	FLAG_ARM_LIBHF.
    
    Signed-off-by: Steve McIntyre <steve.mcintyre@linaro.org>
    Reviewed-by: Carlos O'Donell <carlos@systemhalted.org>

diff --git a/elf/cache.c b/elf/cache.c
index db8b9fa..3336bab 100644
--- a/elf/cache.c
+++ b/elf/cache.c
@@ -94,6 +94,10 @@ print_entry (const char *lib, int flag, unsigned int osversion,
       break;
     case FLAG_MIPS64_LIBN64:
       fputs (",64bit", stdout);
+      break;
+    case FLAG_ARM_LIBHF:
+      fputs (",hard-float", stdout);
+      break;
     case 0:
       break;
     default:
diff --git a/sysdeps/generic/ldconfig.h b/sysdeps/generic/ldconfig.h
index ef3f4b9..a805284 100644
--- a/sysdeps/generic/ldconfig.h
+++ b/sysdeps/generic/ldconfig.h
@@ -34,6 +34,7 @@
 #define FLAG_POWERPC_LIB64	0x0500
 #define FLAG_MIPS64_LIBN32	0x0600
 #define FLAG_MIPS64_LIBN64	0x0700
+#define FLAG_ARM_LIBHF		0x0900
 
 /* Name of auxiliary cache.  */
 #define _PATH_LDCONFIG_AUX_CACHE "/var/cache/ldconfig/aux-cache"
